add control_c384gdas_wav and control_atmwav tests

aliabdolali opened the pull request:
This pull request updates the control tests:
- control_atmwav (modified)
- control_c384gdas_wav (added)
- The WW3 model is updated with the top of NOAA/Develop
- https://github.com/NOAA-EMC/WW3/commit/3740ba99caaa2822b5098bf32faeaf199a710a51

The new grids, mod_defs and rmp_src* for gfsv16 are added to the input-data-20210614/WW3_input_data_20210621 on hera and orion.

new baselines are needed.

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
-  cheyenne.intel, skipped. machine down for week
-  cheyenne.gnu, skipped. machine down for week
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [x] CI: https://github.com/ufs-community/ufs-weather-model/actions/runs/1079472671

JessicaMeixner-NOAA commented on tests/parm/control_gdas_wav.nml.IN:
```diff
@@ -0,0 +1,338 @@
+  prautco      = 0.00015,0.00015
+  lgfdlmprad   = .true.
+  effr_in      = .true.
+  cplwav       = .true.
```
Instead of adding an entire new file, can cplwav just be made a variable input?  I think that helps have fewer files to change as we move forward.

DeniseWorthen replied:
I agree w/ Jessica here. If the only difference between ``control_gdas.nml.IN`` and ``control_gdas_wav.nml.IN`` is the flip of the ``cplwav`` to true, you should use the existing ``control_gdas.nml.IN`` by introducing the ``@[CPLWAV ]`` variable.

aliabdolali replied:
Done

DeniseWorthen commented:
@aliabdolali Please remove the additional ``control_gdas_wav.nml.IN`` and instead use a variable in the default test ``control_gdas.nml.IN``

```
+  cplwav       = @[CPLWAV]
```

CPLWAV is false by default, so in your new test ``control_c384gdas_wav`` set CPLWAV=.true. and change the namelist name that the test uses.

aliabdolali commented:
@DeniseWorthen 
I am working on it, thanks

aliabdolali commented:
@DeniseWorthen I did the suggested changes and tested them.

aliabdolali commented on tests/tests/control_c384gdas_wav:
```diff
@@ -0,0 +1,132 @@
+export CPL=.true.
+export CPLWAV=.true.
+export NEMS_CONFIGURE="nems.configure.leapfrog_atm_wav.IN"
+export CPLWAV=.T.
```
@DeniseWorthen can I remove this? 
`export CPLWAV=.T. `
`as export CPLWAV=.true.` is there

DeniseWorthen replied:
yes, you only need it once. The preference going forward is to use the full word .true. rather than the T/F.

JessicaMeixner-NOAA replied:
CPLWAV is still there twice.

aliabdolali replied:
We investigated it before, removing either of them will break the test. so we kept it.

DeniseWorthen commented:
changes look good.

DeniseWorthen commented:
@aliabdolali We may be moving the commit date for this PR up, depending on the status of Moorthi's PR. Have you added the WW3_input_data_20210518 to additional platforms ? 

Which platforms do you not have access to that you would need me to do that?

aliabdolali commented:
@DeniseWorthen I was waiting for ufs-weather-model team to give me the time to start running final test and add log files. I have the inputs on hera and can add them to orion and run on these 2 platforms.
I also got WCOSS access recently, but have not run anything on it, and probably do not have write permission? 
Could you tell me if this is the time to start, so I can copy inputs to orion, and run on hera and orion with intel?

aliabdolali commented:
@DeniseWorthen On another note, since we have two new tests, do I need to create a new baseline?
rt.sh -c -e rt.conf?

DeniseWorthen commented:
We need to have the data added across all platforms before we begin testing but we should do it ahead of time so we're ready to go. I won't know exactly when we can start the commit process until later today---it depends on how far Moorthi gets on resolving the issues with his PR. 

I will add the needed WW3 input across platforms since you have limited access. There is no wcoss-dell access right now so that will need to wait. 

For the new baseline, yes, we will need to create a new baseline for that single test. Since you're updating WW3 also it might be best to create all new baselines even though you expect no change from the WW3 change alone.

aliabdolali commented:
@DeniseWorthen I just transferred the inputs to Orion
/work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20210614/WW3_input_data_20210621

DeniseWorthen commented:
@aliabdolali Please update the BL_DATE in rt.sh to today's date and commit that change. We we will use the auto-BL creation feature; WCOSS needs to be run manually but we will need to wait for access.

BrianCurtis-NOAA commented:
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/661779909/20210623200042/ufs-weather-model
Please make changes and add the following label back:
hera-gnu-BL

BrianCurtis-NOAA commented:
Machine: hera
Compiler: intel
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/661779909/20210623200529/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_293168
Test control_c384gdas_wav 081 failed failed 
Test control_c384gdas_wav 081 failed in run_test failed 
Test control_atmwav 080 failed in check_result failed 
Test control_atmwav 080 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-BL

DeniseWorthen commented:
@aliabdolali Please check the run directory  ``/scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_293168`` for the failures. The 384gdas test failed with 
```
312:
312:                     *** WAVEWATCH III Multi-grid shell ***
312:                =================================================
312:
321:
321: EXTCDE MPI_ABORT, IEXIT=  2000
```

The control_atmwav test failed because it did not find these files:
```
 Moving out_grd.glo_1deg .........NOT OK. Missing  /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_293168/control_atmwav/out_grd.glo_1deg
 Moving out_pnt.points .........NOT OK. Missing  /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_293168/control_atmwav/out_pnt.points
```

DeniseWorthen commented:
@aliabdolali Let me know when you have finished your fixes. I see you've made a change for the control_atmwav failure.

aliabdolali commented:
@DeniseWorthen bot tests are fixed now.
for the c384gdas_wave test, I had to revert `export CPLWAV=.T.`, otherwise, the mod_def files could not be copied over. 
Where can we fix it? in fv3_conf?
I did not touch other parts of the code, to avoid crashes for other CPLWAV cases.
In the future, if we fix `export CPLWAV=.true.`, then we can use it.

DeniseWorthen commented:
Are you saying that you needed cplwav to be ".true." and not ".T."? We can fix this tomorrow. The  baselines for the remaining tests completed so I can move those and we'll just create the two failed tests manually.

aliabdolali commented:
@DeniseWorthen 
no, there is no problem when it is `cplwav=.T.`, but fails if I use `cplwav=.true.`
The test was running fine until yesterday with `cplwav=.T.`
From a conversation we had yesterday, you mentioned .true. is preferred to .T., so I removed `cplwav=.T.`
And the problem is that it does not copy mod_def files from the input directory. My guess is that we need a fix somewhere in _test/fv3_conf_ to handle **T** or **true** equally. Does it make sense?

DeniseWorthen commented on tests/tests/control_atmwav:
```diff
@@ -48,8 +48,14 @@ export LIST_FILES="sfcf000.nc \
                    RESTART/sfc_data.tile6.nc \
-                   out_grd.glo_1deg \
-                   out_pnt.points"
+                   20210322.090000.out_grd.glo_1deg \
```
Are these restart files at different hours or output? If they're restarts, we generally compare only the final restart.

aliabdolali replied:
out_grd* files are gridded outputs, and out_pnt* files are point outputs.

DeniseWorthen commented:
Yes, I realize that I asked you to switch to true. We should use the .T. because that is how the fv3_conf expects it. Sorry about that.

For the files to compare, you're not comparing any wave restart file, only wave output?

aliabdolali commented:
@DeniseWorthen for the current setting, I have used the ww3_multi template, which does not write restart. @JessicaMeixner-NOAA do we want the restart files? then I can modify it and add the restart at last time step to the comparison list. It is a simple change.

JessicaMeixner-NOAA commented:
@aliabdolali see these lines: 
https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/parm/ww3_multi.inp.IN#L324-L325
if you set the variables as needed you should be able to write restarts if desired.

DeniseWorthen commented:
Two items:

One, I do think we want the comparison files to be the  restart files, and not the output files. 

Second, the tests that you are adding/modifying are not starting from a WW3 ICs, is that correct? In other words, WW3 starts "cold" in these two tests.

aliabdolali commented:
@DeniseWorthen 
I can take out the output files from the list, just let me know.
Yes, both start from cold.

DeniseWorthen commented:
Great. Could you remove the output files from comparison but add the restart files (making whatever change @JessicaMeixner-NOAA is suggesting to create the restart) and then go ahead and create the baselines on hera for the two failed tests. I've copied all the other baselines already, so once we add those two I can add the auto-RT label.

aliabdolali commented:
@DeniseWorthen 
Both tests are updated and the base lines are in
 /scratch1/NCEPDEV/stmp4/Ali.Abdolali/FV3_RT/REGRESSION_TEST_INTEL/
Do I need to copy them or you will take care of them?

DeniseWorthen commented:
I will copy and then add the auto-rt label for hera.intel.

DeniseWorthen commented:
@aliabdolali Would you please commit a change in rt.sh. Your RT just passed so it won't impact this PR, but we want to set it temporarily to batch on hera.

```
diff --git a/tests/rt.sh b/tests/rt.sh
index f770a1d..9d93855 100755
--- a/tests/rt.sh
+++ b/tests/rt.sh
@@ -237,7 +237,7 @@ elif [[ $MACHINE_ID = hera.* ]]; then
   ECFLOW_START=/scratch1/NCEPDEV/nems/emc.nemspara/soft/miniconda3/bin/ecflow_start.sh
   ECF_PORT=$(( $(id -u) + 1500 ))

-  QUEUE=debug
+  QUEUE=batch
   COMPILE_QUEUE=batch
```

aliabdolali commented:
> @aliabdolali Would you please commit a change in rt.sh. Your RT just passed so it won't impact this PR, but we want to set it temporarily to batch on hera.
> 
> ```
> diff --git a/tests/rt.sh b/tests/rt.sh
> index f770a1d..9d93855 100755
> --- a/tests/rt.sh
> +++ b/tests/rt.sh
> @@ -237,7 +237,7 @@ elif [[ $MACHINE_ID = hera.* ]]; then
>    ECFLOW_START=/scratch1/NCEPDEV/nems/emc.nemspara/soft/miniconda3/bin/ecflow_start.sh
>    ECF_PORT=$(( $(id -u) + 1500 ))
> 
> -  QUEUE=debug
> +  QUEUE=batch
>    COMPILE_QUEUE=batch
> ```

Done

BrianCurtis-NOAA commented:
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/661779909/20210624173009/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_33278
Test control_c384gdas_wav 077 failed failed 
Test control_c384gdas_wav 077 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

DeniseWorthen commented:
On gaea, the err file shows:
```
srun: error: nid01137: tasks 378-383,385-411: Killed
srun: Terminating job step 268839422.0
srun: error: nid01135: tasks 310-311: Killed
slurmstepd: error: *** STEP 268839422.0 ON nid00728 CANCELLED AT 2021-06-24T18:24:16 ***
```

I'm re-trying creating that baseline manually.

BrianCurtis-NOAA commented:
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/661779909/20210624114512/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_65121
Test control_c384gdas_wav 079 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-BL

BrianCurtis-NOAA commented:
Machine: cheyenne
Compiler: gnu
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/661779909/20210624113012/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_20543
Baseline creation and move successful
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/661779909/20210624115814/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_38934
Test control_thompson_debug 020 failed in run_test failed 
Test control_ras_debug 024 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-gnu-BL

DeniseWorthen commented:
On Cheyenne, the control_ras_debug timed out w/o even starting up. The control_thompson_debug failed with:

```
137:mp_thompson_post_run: ttendlim applied 2/4064 times at timestep 1
144:
144:Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
144:
```

DeniseWorthen commented:
@aliabdolali I was able to compile and run the control_c384gdas_wav on gaea but it timed out before the wave restarts were written. Are you sure you've set the resources correctly for all platforms? Can the test length be shortened ? 

For Cheyenne Intel, I copied your run directory and resubmitted. The job again failed with the same error: 
MPT: shepherd terminated: r14i5n23.ib0.cheyenne.ucar.edu - job aborting

I believe that on Cheyenne, this usually indicates a system issue but I'm surprised running a second time would not have resolved that.

The Cheyenne gnu failure must be unrelated to your PR since we don't run ATMW with GNU.

aliabdolali commented:
@DeniseWorthen 
I did not have access to Gaea, but we might need to adduct something for it in tests/tests/control_c384gdas_wav

```
export TASKS=480
if [[ $MACHINE_ID = cheyenne.* ]]; then
  export TPN=36
elif [[ $MACHINE_ID = hera.* ]]; then
  export TPN=20
  export THRD=2
elif [[ $MACHINE_ID = wcoss_cray ]]; then
  export TPN=6
else
  export TPN=12
fi
```

Do you have any suggestions?

BrianCurtis-NOAA commented:
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/661779909/20210624173014/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_81717
Test control_c384gdas_wav 078 failed failed 
Test control_c384gdas_wav 078 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

DeniseWorthen commented:
@aliabdolali I do not enough about the resources this test might require. I don't think platform-specific adjustments like the one you show are made in most tests. You can look in default_vars for the resources available on each platform.

I believe you said you have tested on both Orion and Hera. Have you been able to test on either dell-p3 or jet? We do not run the wave tests on wcoss-cray. @JessicaMeixner-NOAA do you have any insight to offer here?

JessicaMeixner-NOAA commented:
So this test for the wave side is a little bit memory intensive so, running with 2 threads on all the machines is likely a good idea, especially if it's not running to completion.  If you've optimized it and checked that you are in balance on both hera and orion, you can generally make good educated guesses based on the resources @DeniseWorthen pointed you to for the other machines.

DeniseWorthen commented:
To summarize at this point, the control_c384gdas_wav baseline has not been created on Cheyenne.intel, Jet.intel and Gaea.intel but otherwise the develop-20210623 is fully populated. 

No WCOSS has been run yet (but we don't run wave on cray). 

Hera (intel,gnu), Cheyenne.gnu and Orion are done.

The control_c384gdas is not run on Gaea, so there may be a reason not to run the control_c384gdas_wav on Gaea.

JessicaMeixner-NOAA commented:
If control_c384gdas is not run on Gaea, seems reasonable to not run control_c384gdas_wav on Gaea although I don't know why the atm-only is not run on gaea to know for sure.  Since @aliabdolali does not have gaea access, I can try to run some tests to help figure that out, but that's not likely going to happen quickly as others will want in terms of holding up the queue.

DeniseWorthen commented:
@aliabdolali Jun noticed that the job cards on gaea and jet may need to be changed to allow the nodes and tasks-per-node setting. WCOSS is open now, and I will transfer the new WW3 data and start the baseline creations there in the meantime.

JessicaMeixner-NOAA commented:
@DeniseWorthen I can help with the gaea testing.  I theoretically have access to jet but not sure if I have projects on there right now.  Should I wait for updates to the gaea job card or go ahead and work on testing/optimizing it there?  Did we ever find out why the control_c384gdas is not run on gaea?

DeniseWorthen commented:
Thanks @JessicaMeixner-NOAA let's hold off for the time being. Jun has tests running on Gaea and Jet. I believe the existing control_c384gdas is not being run on Gaea for the same reason.

junwang-noaa commented:
@DeniseWorthen @aliabdolali We need following changes on gaea: tests/fv3_conf/fv3_slurm.IN_gaea
```
 @@ -5,7 +5,8 @@
#SBATCH --account=@[ACCNR]
 #SBATCH --qos=@[QUEUE]
 #SBATCH --clusters=@[PARTITION]
-#SBATCH --ntasks=@[TASKS]
+#SBATCH --nodes=@[NODES]
+#SBATCH --ntasks-per-node=@[TPN]

@@ -28,7 +29,7 @@ export NC_BLKSZ=1M
 sync && sleep 1

-srun ./fv3.exe
+srun --label -n @[TASKS] ./fv3.exe

```

With the changes, you can turn on control_c384 and control_c384gdas and control_c384gdas_wav is running to completion.

junwang-noaa commented:
Sorry, forgot the change in tests/default_vars.sh:

```
@@ -242,7 +242,7 @@ elif [[ $MACHINE_ID = gaea.* ]]; then

   TASKS_dflt=150 ; TPN_dflt=36 ; INPES_dflt=3 ; JNPES_dflt=8
   TASKS_thrd=78  ; TPN_thrd=18 ; INPES_thrd=3 ; JNPES_thrd=4
-  TASKS_c384=480 ; TPN_c384=36 ; INPES_c384=12 ; JNPES_c384=6
+  TASKS_c384=480 ; TPN_c384=18 ; INPES_c384=12 ; JNPES_c384=6
```

DeniseWorthen commented:
@aliabdolali Please make the following change to rt.conf. This will turn on the ``control_c384gdas`` test on gaea and switch the two atm+wav tests to 32bit. 

Switching to 32bit will mean we will need to create new baselines on orion and hera for the two affected tests because answers will change. Can you do that?

I will do gaea.

```
diff --git a/tests/rt.conf b/tests/rt.conf
index cbbf598..85777f7 100644
--- a/tests/rt.conf
+++ b/tests/rt.conf
@@ -44,7 +44,7 @@ RUN     | control_CubedSphereGrid
 RUN     | control_wrtGauss_netcdf_parallel                                                                                        |                                         | fv3 |
 RUN     | control_c192                                                                                                            |                                         | fv3 |
 RUN     | control_c384                                                                                                            | - gaea.intel wcoss_cray                 | fv3 |
-RUN     | control_c384gdas                                                                                                        | - gaea.intel wcoss_cray                 | fv3 |
+RUN     | control_c384gdas                                                                                                        | - wcoss_cray                            | fv3 |
 RUN     | control_stochy                                                                                                          |                                         | fv3 |
 RUN     | control_ca                                                                                                              |                                         | fv3 |
 RUN     | control_lndp                                                                                                            |                                         | fv3 |
@@ -185,6 +185,6 @@ RUN     | datm_cdeps_debug_cfsr
 # ATM-WAV tests                                                                                                                                                                   #
 ###################################################################################################################################################################################

-COMPILE | APP=ATMW SUITES=FV3_GFS_v16                                                                                           | - wcoss_cray  | fv3 |
+COMPILE | APP=ATMW SUITES=FV3_GFS_v16 32BIT=Y                                                                                   | - wcoss_cray  | fv3 |
 RUN     | control_atmwav                                                                                                        | - wcoss_cray  | fv3 |
 RUN     | control_c384gdas_wav                                                                                                  | - wcoss_cray  | fv3 |
```

junwang-noaa commented:
Well, I had a typo in jet job card, I have to resubmit the tests. Below are the code changes, similar to those on gaea:

```
tests/fv3_conf/fv3_slurm.IN_jet
@@ -4,7 +4,8 @@
 #SBATCH --account=@[ACCNR]
 #SBATCH --partition=@[PARTITION]
 #SBATCH --qos=@[QUEUE]
-#SBATCH --ntasks=@[TASKS]
+#SBATCH --nodes=@[NODES]
+#SBATCH --ntasks-per-node=@[TPN]
 #SBATCH --time=@[WLCLK]
 #SBATCH --job-name="@[JBNME]"
 ### #SBATCH --exclusive
@@ -32,7 +33,7 @@ export PSM_SHAREDCONTEXTS=1
 # Avoid job errors because of filesystem synchronization delays
 sync && sleep 1

-srun ./fv3.exe
+srun --label -n @[TASKS] ./fv3.exe

tests/default_vars.sh
@@ -190,7 +190,7 @@ elif [[ $MACHINE_ID = jet.* ]]; then

   TASKS_dflt=150 ; TPN_dflt=24 ; INPES_dflt=3 ; JNPES_dflt=8
   TASKS_thrd=78  ; TPN_thrd=12 ; INPES_thrd=3 ; JNPES_thrd=4
-  TASKS_c384=480 ; TPN_c384=24 ; INPES_c384=12 ; JNPES_c384=6
+  TASKS_c384=480 ; TPN_c384=12 ; INPES_c384=12 ; JNPES_c384=6

```

aliabdolali commented:
@DeniseWorthen 
here is the path to the new bases for two new tests with _32BIT=Y_ on hera
_/scratch1/NCEPDEV/stmp4/Ali.Abdolali/FV3_RT/REGRESSION_TEST_INTEL_
could you copy them over.
orion is still running...

DeniseWorthen commented:
yes I will copy.

DeniseWorthen commented:
@aliabdolali Can you make this change for Cheyenne:
```
diff --git a/tests/tests/control_c384gdas_wav b/tests/tests/control_c384gdas_wav
index a51a5fe..0221b5c 100644
--- a/tests/tests/control_c384gdas_wav
+++ b/tests/tests/control_c384gdas_wav
@@ -56,7 +56,7 @@ export_fv3

 export TASKS=480
 if [[ $MACHINE_ID = cheyenne.* ]]; then
-  export TPN=36
+  export TPN=18
 elif [[ $MACHINE_ID = hera.* ]]; then
   export TPN=20
   export THRD=2
```

aliabdolali commented:
@DeniseWorthen 
here are the bases on orion 
_/work/noaa/stmp/aabdolal/stmp/aabdolal/FV3_RT/REGRESSION_TEST_INTEL_

DeniseWorthen commented:
The control_c384gdas_wav test is still failing on Cheyenne.intel with error:

MPT: shepherd terminated: r3i6n22.ib0.cheyenne.ucar.edu - job aborting

junwang-noaa commented:
Did the test actually start to run on cheyenne?

On Fri, Jun 25, 2021 at 1:32 PM Denise Worthen ***@***.***>
wrote:

> The control_c384gdas_wav test is still failing on Cheyenne.intel with
> error:
>
> MPT: shepherd terminated: r3i6n22.ib0.cheyenne.ucar.edu - job aborting
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-868723836>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TJOVGHPICZA33TZWLDTUS4SDANCNFSM46C2ZPKQ>
> .
>

DeniseWorthen commented:
It dies in the initialization. It never actually starts to run. The last thing I see in the PET000 log is 

```
20210625 172243.500 INFO             PET000 bf FieldBundleRegridStore
20210625 172243.500 INFO             PET000 calling into wrtFB(01,01) FieldBundleRegridStore()....
20210625 172246.776 INFO             PET000 af FieldBundleRegridStore
20210625 172246.776 INFO             PET000 ... returned from wrtFB(01,01) FieldBundleRegridStore().
20210625 172246.776 INFO             PET000 bf FieldBundleRegridStore
20210625 172246.776 INFO             PET000 calling into wrtFB(02,01) FieldBundleRegridStore()....
20210625 172249.048 INFO             PET000 af FieldBundleRegridStore
```

In err

```
0: af create fcst fieldbundle, name=atmrc=           0
0: af create fcst fieldbundle, name=sfc_nearest_stodrc=           0
0: af create fcst fieldbundle, name=sfc_bilinearrc=           0
0: in fcst,init total time:    15.0761558637023
MPT: shepherd terminated: r3i6n22.ib0.cheyenne.ucar.edu - job aborting
```

DeniseWorthen commented:
On jet, the control_c384gdas_wav job times out before it writes the wave restarts. All the other files are written OK.

climbfuji commented:
> It dies in the initialization. It never actually starts to run. The last thing I see in the PET000 log is
> 
> ```
> 20210625 172243.500 INFO             PET000 bf FieldBundleRegridStore
> 20210625 172243.500 INFO             PET000 calling into wrtFB(01,01) FieldBundleRegridStore()....
> 20210625 172246.776 INFO             PET000 af FieldBundleRegridStore
> 20210625 172246.776 INFO             PET000 ... returned from wrtFB(01,01) FieldBundleRegridStore().
> 20210625 172246.776 INFO             PET000 bf FieldBundleRegridStore
> 20210625 172246.776 INFO             PET000 calling into wrtFB(02,01) FieldBundleRegridStore()....
> 20210625 172249.048 INFO             PET000 af FieldBundleRegridStore
> ```
> 
> In err
> 
> ```
> 0: af create fcst fieldbundle, name=atmrc=           0
> 0: af create fcst fieldbundle, name=sfc_nearest_stodrc=           0
> 0: af create fcst fieldbundle, name=sfc_bilinearrc=           0
> 0: in fcst,init total time:    15.0761558637023
> MPT: shepherd terminated: r3i6n22.ib0.cheyenne.ucar.edu - job aborting
> ```

Try to find out which MPI task is crashing, then look into that MPI tasks' log files (PET log, stdout and stderr)

junwang-noaa commented:
I'd suggest reducing the 384 wave test forecast length to 9hrs.


On Fri, Jun 25, 2021 at 1:43 PM Denise Worthen ***@***.***>
wrote:

> On jet, the job times out before it writes the wave restarts. All the
> other files are written OK.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-868730270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TNBXIDA3B3A2M2KQQDTUS54VANCNFSM46C2ZPKQ>
> .
>

DeniseWorthen commented:
Reducing the forecast time would mean re-creating this test's baseline on all platforms again, which would be painful but doable. 

I can't find anything which indicates why the job is failing on cheyenne. The run directory is  /glade/scratch/worthen/FV3_RT/rt_31708

DeniseWorthen commented:
@junwang-noaa I think the best solution at this point might be to remove this test on cheyenne.intel and jet.intel until we can debug whatever the issue is on cheyenne as well as reduce the forecast time. I still have Dell to finish (the verification was running when they took Mars off-line).

junwang-noaa commented:
That is fine with me.

On Fri, Jun 25, 2021 at 3:56 PM Denise Worthen ***@***.***>
wrote:

> @junwang-noaa <https://github.com/junwang-noaa> I think the best solution
> at this point might be to remove this test on cheyenne.intel and jet.intel
> until we can debug whatever the issue is on cheyenne as well as reduce the
> forecast time. I still have Dell to finish (the verification was running
> when they took Mars off-line).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-868800890>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TLMRPVLR46WUP7BZTTTUTNPDANCNFSM46C2ZPKQ>
> .
>

DeniseWorthen commented:
@aliabdolali Please commit the following change to rt.conf
```
diff --git a/tests/rt.conf b/tests/rt.conf
index 85777f7..91e278c 100644
--- a/tests/rt.conf
+++ b/tests/rt.conf
@@ -185,6 +185,6 @@ RUN     | datm_cdeps_debug_cfsr
 # ATM-WAV tests                                                                                                                                                                   #
 ###################################################################################################################################################################################

-COMPILE | APP=ATMW SUITES=FV3_GFS_v16 32BIT=Y                                                                                   | - wcoss_cray  | fv3 |
-RUN     | control_atmwav                                                                                                        | - wcoss_cray  | fv3 |
-RUN     | control_c384gdas_wav                                                                                                  | - wcoss_cray  | fv3 |
+COMPILE | APP=ATMW SUITES=FV3_GFS_v16 32BIT=Y                                                                                   | - wcoss_cray                          | fv3 |
+RUN     | control_atmwav                                                                                                        | - wcoss_cray                          | fv3 |
+RUN     | control_c384gdas_wav                                                                                                  | - wcoss_cray jet.intel cheyenne.intel | fv3 |
```

climbfuji commented:
> That is fine with me.
> […](#)

I disagree. If we continue doing this we will run a selection of regression tests, some on this system, some on another. We need to take the time to figure this out. In this case, it was easy. I got it to run by using more nodes / less tasks per node:
```
#!/bin/sh -l
#PBS -o out.pbs
#PBS -e err.pbs
#PBS -N rt_31708_002
#PBS -A P48503002
#PBS -q economy
#PBS -l select=40:ncpus=36:mpiprocs=36
#PBS -l walltime=00:30:00

set -eux
echo -n " $( date +%s )," >  job_timestamp.txt

set +x
source ./module-setup.sh
module use $( pwd -P )
module load modules.fv3
module list

set -x

echo "Model started:  " `date`

export MPI_TYPE_DEPTH=20
export OMP_STACKSIZE=512M
export OMP_NUM_THREADS=1
export ESMF_RUNTIME_COMPLIANCECHECK=OFF:depth=4

# Avoid job errors because of filesystem synchronization delays
sync && sleep 1

mpiexec_mpt -p %g: -ppn 12 -np 480 ./fv3.exe >out 2>err

echo "Model ended:    " `date`
echo -n " $( date +%s )," >> job_timestamp.txt
```

DeniseWorthen commented:
My concern was that if this PR stretched into Monday, it was going to have a cascading impact on a very full commit queue, some of which have been waiting quite some time. In hindsight I should have been more aware that Ali had limited access to other machines and would not be able to test the platform-specific changes he needed to make. 

I'm willing to try to get the baseline for this final test created on cheyenne.intel over the weekend, since you've spent your time to debug it. 

In terms of the forecast length, the WW3 model is currently writing restarts every 3hrs, but only needs the restart at the final time. I can test changing the restart frequency on jet and see if it can complete the test.

climbfuji commented:
> My concern was that if this PR stretched into Monday, it was going to have a cascading impact on a very full commit queue, some of which have been waiting quite some time. In hindsight I should have been more aware that Ali had limited access to other machines and would not be able to test the platform-specific changes he needed to make.
> 
> I'm willing to try to get the baseline for this final test created on cheyenne.intel over the weekend, since you've spent your time to debug it.
> 
> In terms of the forecast length, the WW3 model is currently writing restarts every 3hrs, but only needs the restart at the final time. I can test changing the restart frequency on jet and see if it can complete the test.

Let me know if I can help. I will have a bit of time and internet access on the weekend to create baselines on cheyenne and/or jet and verify against it.

JessicaMeixner-NOAA commented:
@DeniseWorthen while writing fewer restarts should help, my guess is that more resources is likely what is needed.

junwang-noaa commented:
I agree with Denise. We have many commits ready in the commit queue, if for
this PR, there are still some issues we can not resolve today, we have to
1) turn off that test on cheyenne and jet and have a follow up PR 2) stop
this PR and move to the next PR. We can not spend days trying to figure
things out while holding other commits. I don't think choosing 1) means
that we will run selection of the tests on each platform, as Denise
mentioned we can have follow up PR to fix  issues on Cheyenne and jet.
Please don't delay the whole commit queue.

On Fri, Jun 25, 2021 at 10:22 PM Jessica Meixner ***@***.***>
wrote:

> @DeniseWorthen <https://github.com/DeniseWorthen> while writing fewer
> restarts should help, my guess is that more resources is likely what is
> needed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-868929884>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TJNB5ZBCFOMRDPQTLDTUU2UVANCNFSM46C2ZPKQ>
> .
>

BrianCurtis-NOAA commented:
Machine: cheyenne
Compiler: intel
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/661779909/20210626120011/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_53795
Test control_thompson_debug 064 failed in run_test failed 
Test control_decomp 017 failed in run_test failed 
Test control_ugwpv1 032 failed in run_test failed 
Test control_ras 033 failed in run_test failed 
Test control_flake 031 failed in run_test failed 
Test control_thompson 034 failed in run_test failed 
Test control_noahmp 036 failed in run_test failed 
Test fv3_rap 046 failed in run_test failed 
Test fv3_hrrr 047 failed in run_test failed 
Test fv3_rrfs_v1beta 048 failed in run_test failed 
Test control_thompson_no_aero 035 failed in run_test failed 
Test datm_bulk_cfsr 076 failed in run_test failed 
Test datm_control_cfsr 073 failed in run_test failed 
Test datm_control_gefs 075 failed in run_test failed 
Test datm_bulk_gefs 077 failed in run_test failed 
Test datm_mx025_gefs 079 failed in run_test failed 
Test datm_mx025_cfsr 078 failed in run_test failed 
Test datm_debug_cfsr 080 failed in run_test failed 
Test fv3_rrfs_v1alpha 045 failed in run_test failed 
Test fv3_gsd 044 failed in run_test failed 
Test fv3_HAFS_v0_hwrf_thompson 049 failed in run_test failed 
Test control_c384gdas 024 failed in run_test failed 
Test control_wrtGauss_netcdf_parallel 021 failed in run_test failed 
Test regional_quilt 039 failed in run_test failed 
Test regional_quilt_2threads 040 failed in run_test failed 
Test control_ca 026 failed in run_test failed 
Test control_lndp 027 failed in run_test failed 
Test control 016 failed in run_test failed 
Test control_lheatstrg 028 failed in run_test failed 
Test regional_control 037 failed failed 
Test regional_control 037 failed in run_test failed 
Test fv3_HAFS_v0_hwrf_thompson_debug 071 failed failed 
Test fv3_HAFS_v0_hwrf_thompson_debug 071 failed in run_test failed 
Test regional_quilt_RRTMGP 043 failed in run_test failed 
Test control_CubedSphereGrid 020 failed in run_test failed 
Test control_rrtmgp 030 failed in run_test failed 
Test fv3_esg_HAFS_v0_hwrf_thompson_debug 072 failed in run_test failed 
Test fv3_esg_HAFS_v0_hwrf_thompson 050 failed in run_test failed 
Test control_merra2 029 failed in run_test failed 
Test control_stochy 025 failed in run_test failed 
Test control_2threads 018 failed in run_test failed 
Test control_c192 022 failed in run_test failed 
Test regional_quilt_hafs 041 failed in run_test failed 
Test regional_quilt_netcdf_parallel 042 failed in run_test failed 
Test control_c384 023 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-RT

DeniseWorthen commented:
The cheyenne.intel auto-RT job had several compile failures (1,2,15,16,17) and several jobs failed with  ``line          363 NetCDF: HDF error`` in the err file. I couldn't find any disk quota errors but that is all I can think of. I removed a lot of old rt_* directories and ran it manually. I still had a single test failure (a datm test) but was able to re-run that test and pass.

As currently configured, gaea,orion,dell_p3 and cheyenne take ~25mins for the test.  On jet, I was able to get ~25min by reducing the forecast time to 9 hrs so I've kept it off the list for now.

I'm concerned that running this close to the wall clock on 4 platforms will lead to frequent RT failures. I think we should consider holding this PR back until we can optimize the run times better.

junwang-noaa commented:
Is the run time for the C384gdas test OK? We may need to consider
increasing the number of tasks for wave. I am a little concerned about the
number of tasks for wave. I remember in high resolution C768L127 gfsv16
runs wave takes about 1/4 of total number fv3atm tasks. Now in this
C384L127 gdas wave test, wave uses 1/2 of the atm tasks (312 vs168), the
total run time on gaea with and without wave is: 780s vs 1472s.
@aliabdolali <https://github.com/aliabdolali> Is the wave configuration in
this test the same as opn gdas forecast?

On Sat, Jun 26, 2021 at 7:31 PM Denise Worthen ***@***.***>
wrote:

> The cheyenne.intel auto-RT job had several compile failures (1,2,15,16,17)
> and several jobs failed with line 363 NetCDF: HDF error in the err file.
> I couldn't find any disk quota errors but that is all I can think of. I
> removed a lot of old rt_* directories and ran it manually. I still had a
> single test failure (a datm test) but was able to re-run that test and pass.
>
> As currently configured, gaea,orion,dell_p3 and cheyenne take ~25mins for
> the test. On jet, I was able to get ~25min by reducing the forecast time to
> 9 hrs so I've kept it off the list for now.
>
> I'm concerned that running this close to the wall clock on 4 platforms
> will lead to frequent RT failures. I think we should consider holding this
> PR back until we can optimize the run times better.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-869073778>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TKN2UNMN4CYZA26RT3TUZPNZANCNFSM46C2ZPKQ>
> .
>

aliabdolali commented:
> Is the run time for the C384gdas test OK? We may need to consider increasing the number of tasks for wave. I am a little concerned about the number of tasks for wave. I remember in high resolution C768L127 gfsv16 runs wave takes about 1/4 of total number fv3atm tasks. Now in this C384L127 gdas wave test, wave uses 1/2 of the atm tasks (312 vs168), the total run time on gaea with and without wave is: 780s vs 1472s. @aliabdolali <https://github.com/aliabdolali> Is the wave configuration in this test the same as opn gdas forecast?
> […](#)
> On Sat, Jun 26, 2021 at 7:31 PM Denise Worthen ***@***.***> wrote: The cheyenne.intel auto-RT job had several compile failures (1,2,15,16,17) and several jobs failed with line 363 NetCDF: HDF error in the err file. I couldn't find any disk quota errors but that is all I can think of. I removed a lot of old rt_* directories and ran it manually. I still had a single test failure (a datm test) but was able to re-run that test and pass. As currently configured, gaea,orion,dell_p3 and cheyenne take ~25mins for the test. On jet, I was able to get ~25min by reducing the forecast time to 9 hrs so I've kept it off the list for now. I'm concerned that running this close to the wall clock on 4 platforms will lead to frequent RT failures. I think we should consider holding this PR back until we can optimize the run times better. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#620 (comment)](https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-869073778)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AI7D6TKN2UNMN4CYZA26RT3TUZPNZANCNFSM46C2ZPKQ> .

@junwang-noaa I optimized the number of cores on hera, and from a conversation with you on this, it was about 1000 s.

junwang-noaa commented:
@aliabdolali 1000s for this test is good. I have two questions. First, Is the C384gdas_wave using the same wave configuration as opn gdas forecast? second, how did you optimize the number of cores on hera? Can you make similar optimization on orion and dell-p3? We may need to apply that to gaea and cheyenne. As Denise mentioned, the run is very slow on jet, 6hr integration takes 25mins, we may need to reset the number of tasks there too.

DeniseWorthen commented:
I had also noticed that the wave version of the test is using only 24 write tasks per group but the control_c384gdas is using 48. Will this impact the wall time of the test?

aliabdolali commented:
> @aliabdolali 1000s for this test is good. I have two questions. First, Is the C384gdas_wave using the same wave configuration as opn gdas forecast? second, how did you optimize the number of cores on hera? Can you make similar optimization on orion and dell-p3? We may need to apply that to gaea and cheyenne. As Denise mentioned, the run is very slow on jet, 6hr integration takes 25mins, we may need to reset the number of tasks there too.

This is the GFSv16 setup which I think is similar to gdas.
I used https://github.com/ufs-community/ufs-weather-model/wiki/Advanced-Topics-for-developers#profiling-timing-across-components for profiling. 
I will check it on orion and will let you know first.

JessicaMeixner-NOAA commented:
@aliabdolali the GFS and GDAS wave grids are the exact same.  The resources used for each are different however. I have access to gaea and can help on gaea if needed for the optimization.

aliabdolali commented:
@junwang-noaa @DeniseWorthen I merge ufs-weather-model/develop and also updated WW3 version in this PR. I changed the control_c384gdas_wav forecast hour to 9 hr (from 12 hrs) and just wrote restart at the last time step to save the time it was used for the restart dump. I tested it on Orion/hera and it is less than 900 s.

I am trying to test it on WCOSS but I could not use -e option and my job is stuck. Since WCOSS scheduler id not slurm and it is my first time on it, I wonder if I am doing sth wrong there:
./rt.sh -e -c -k -l rt.test
rt.test is just two new wave tests

DeniseWorthen commented:
@aliabdolali That command looks correct. I usually put all the options together but I don't think that matters. I will try on dell-p3 too.

```
./rt.sh -cek -l rt.test >output 2>&1 &
```

I am trying your branch on jet which we were unable to get to finish in the 30m wall time. I see you made a fix to the fv3_slum.IN_jet.

DeniseWorthen commented:
On dell, I got wildly different wall clock times for the two tests:

0] The total amount of wall time                        = 155.081167
[0] The total amount of wall time                        = 1273.204214

That is still over 20min for the gdas test. The jet job is still pending.

DeniseWorthen commented:
The jet job completed, but just barely. The run time was 24.5 minutes, so about what we were getting on other platforms before you reduced the forecast hours to 9.

aliabdolali commented:
> The jet job completed, but just barely. The run time was 24.5 minutes, so about what we were getting on other platforms before you reduced the forecast hours to 9.

@DeniseWorthen the run time < 1000 s if I add 72 more cores to the wave. so from 480 > 552 total (312 atm and 240 wave). 
I tested it on hera, orion and wcoss.

JessicaMeixner-NOAA commented:
@aliabdolali I'm about to run this on gaea, not sure if it matters but I noticed that the branch is not up to date with ufs-weather-model and WW3 is not up to date with its latest develop either.

aliabdolali commented:
> @aliabdolali I'm about to run this on gaea, not sure if it matters but I noticed that the branch is not up to date with ufs-weather-model and WW3 is not up to date with its latest develop either.

Could you try it now? I am going to check on Hera again.

JessicaMeixner-NOAA commented on tests/rt.conf:
```diff
@@ -189,12 +189,14 @@ RUN     | datm_cdeps_debug_cfsr
 
-COMPILE | -DAPP=ATMW -DCCPP_SUITES=FV3_GFS_v16                                                                                  | - wcoss_cray  | fv3 |
-RUN     | control_atmwav                                                                                                        | - wcoss_cray  | fv3 |
+COMPILE | APP=ATMW SUITES=FV3_GFS_v16 32BIT=Y                                                                                   | - wcoss_cray                          | fv3 |
```
@aliabdolali this line needs to be reverted to what it is in the develop branch
as is the model does not compile (which is why I still don't have updates on gaea)

aliabdolali replied:
@JessicaMeixner-NOAA I reverted it. However, 32BIT=Y  was suggested by @DeniseWorthen

JessicaMeixner-NOAA replied:
Adding 32BIT=Y is a great suggestion, but the "APP=ATMW and SUITES=.." is no longer the correct format and the model will not build as it was.   Are we going to use 32bit?   If so, I need to know so I can properly update the test for gaea.

junwang-noaa replied:
I think it is good to have 32BIT=Y as this test is a low resolution test for GFSv16.

aliabdolali replied:
I added 32BIT=Y to it.

JessicaMeixner-NOAA replied:
Does it compile?  The other compile lines with 32BIT use "-D32BIT=ON"?

aliabdolali replied:
Things are changing rapidly. It was working a few weeks ago. I changed it to -D32BIT=ON and tested it.

JessicaMeixner-NOAA replied:
Just trying to help out, since gaea is down and because I was trying to debug the compiler errors I was getting on gaea, I have run on hera as well.  The results are here: /scratch1/NCEPDEV/stmp2/Jessica.Meixner/FV3_RT/rt_203871 @aliabdolali can you confirm this in line with your results from the last time you ran and had things working on hera for the two ATMW tests?

aliabdolali replied:
Thanks Jessica, I checked the log files, run times and the list of wave out_grd and out_pnt and restart files. They look ok. I also ran the whole rt.conf last night on hera, as I synced my branch with the ufs-weather-model and all ran successfully.

JessicaMeixner-NOAA replied:
So it's expected that in both cases that the atm model runs about 2x as fast as the wave model?  This is what I'm seeing in the ESMF_Profile.summary, just want to confirm.

aliabdolali replied:
From a conversation we had with @junwang-noaa, we do not want to allocate more cpu than this to the wave model. otherwise, we might be able to reduce the whole simulation by adding more cores. Or I can reduce the atm cores and add them to the wave model. But again, Jun mentioned in the past we had 1/3 to wave, 2/3 to atm.

junwang-noaa commented:
This is actually my question. Is this expected that the wave model needs to
use the same as or more tasks than that used in global FV3 to get
comparable run time as atm?

On Wed, Jul 21, 2021 at 12:46 PM Ali.Abdolali ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/rt.conf
> <https://github.com/ufs-community/ufs-weather-model/pull/620#discussion_r674202076>
> :
>
> > @@ -189,12 +189,14 @@ RUN     | datm_cdeps_debug_cfsr
>  # ATM-WAV tests                                                                                                                                                                   #
>  ###################################################################################################################################################################################
>
> -COMPILE | -DAPP=ATMW -DCCPP_SUITES=FV3_GFS_v16                                                                                  | - wcoss_cray  | fv3 |
> -RUN     | control_atmwav                                                                                                        | - wcoss_cray  | fv3 |
> +COMPILE | APP=ATMW SUITES=FV3_GFS_v16 32BIT=Y                                                                                   | - wcoss_cray                          | fv3 |
>
> From a conversation we had with @junwang-noaa
> <https://github.com/junwang-noaa>, we do not want to allocate more cpu
> than this to the wave model. otherwise, we might be able to reduce the
> whole simulation by adding more cores. Or I can reduce the atm cores and
> add them to the wave model. But again, Jun mentioned in the past we had 1/3
> to wave, 2/3 to atm.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#discussion_r674202076>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TNHZYQ7IOUKMCOWDATTY4BWNANCNFSM46C2ZPKQ>
> .
>

JessicaMeixner-NOAA commented:
At lower resolutions of the atm model, such as the C96, I do expect to need to use quite a bit for the wave tasks to keep things at the same speed right now. 

For the C384+wav GDAS like test, if the atm is using what it's using in operations, I'd expect the wave model to need about the same too.  But if the atm model is not using as much I would expect that similarly the wave model could get away with less not more.  

I'll report back what I can figure out on gaea.

junwang-noaa commented:
May I ask what the GFSv16 gdas forecast (C384+wav) is using for wave tasks?

On Wed, Jul 21, 2021 at 2:29 PM Jessica Meixner ***@***.***>
wrote:

> At lower resolutions of the atm model, such as the C96, I do expect to
> need to use quite a bit for the wave tasks to keep things at the same speed
> right now.
>
> For the C384+wav GDAS like test, if the atm is using what it's using in
> operations, I'd expect the wave model to need about the same too. But if
> the atm model is not using as much I would expect that similarly the wave
> model could get away with less not more.
>
> I'll report back what I can figure out on gaea.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-884440548>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TI4R4HBHGE42NGD7ZLTY4N2PANCNFSM46C2ZPKQ>
> .
>

aliabdolali commented:
> May I ask what the GFSv16 gdas forecast (C384+wav) is using for wave tasks?
> […](#)
> On Wed, Jul 21, 2021 at 2:29 PM Jessica Meixner ***@***.***> wrote: At lower resolutions of the atm model, such as the C96, I do expect to need to use quite a bit for the wave tasks to keep things at the same speed right now. For the C384+wav GDAS like test, if the atm is using what it's using in operations, I'd expect the wave model to need about the same too. But if the atm model is not using as much I would expect that similarly the wave model could get away with less not more. I'll report back what I can figure out on gaea. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#620 (comment)](https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-884440548)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AI7D6TI4R4HBHGE42NGD7ZLTY4N2PANCNFSM46C2ZPKQ> .

BTW, I can tune the number of pets for atm and wave models (total of 512 cores) if you want @junwang-noaa ? I thought we should keep  312 pets for atm.

junwang-noaa commented:
OK, let's do that. Jobs will be waiting longer in the queue if we request
more tasks (we have a memory requirement too for this C384L127 runs).

On Wed, Jul 21, 2021 at 3:31 PM Ali.Abdolali ***@***.***>
wrote:

> May I ask what the GFSv16 gdas forecast (C384+wav) is using for wave tasks?
> … <#m_8399625657220534341_>
> On Wed, Jul 21, 2021 at 2:29 PM Jessica Meixner *@*.***> wrote: At lower
> resolutions of the atm model, such as the C96, I do expect to need to use
> quite a bit for the wave tasks to keep things at the same speed right now.
> For the C384+wav GDAS like test, if the atm is using what it's using in
> operations, I'd expect the wave model to need about the same too. But if
> the atm model is not using as much I would expect that similarly the wave
> model could get away with less not more. I'll report back what I can figure
> out on gaea. — You are receiving this because you were mentioned. Reply to
> this email directly, view it on GitHub <#620 (comment)
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-884440548>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AI7D6TI4R4HBHGE42NGD7ZLTY4N2PANCNFSM46C2ZPKQ
> .
>
> BTW, I can tune the number of pets for atm and wave models (total of 512
> cores) if you want @junwang-noaa <https://github.com/junwang-noaa> ? I
> thought we should keep 312 pets for atm.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/620#issuecomment-884479416>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TMXUBWERM5MYA2DBKLTY4VB7ANCNFSM46C2ZPKQ>
> .
>

DeniseWorthen commented:
@aliabdolali Please update the baseline date in rt.sh

JessicaMeixner-NOAA commented on tests/default_vars.sh:
```diff
@@ -190,7 +190,7 @@ elif [[ $MACHINE_ID = jet.* ]]; then
   TASKS_dflt=150 ; TPN_dflt=24 ; INPES_dflt=3 ; JNPES_dflt=8
   TASKS_thrd=78  ; TPN_thrd=12 ; INPES_thrd=3 ; JNPES_thrd=4
-  TASKS_c384=480 ; TPN_c384=24 ; INPES_c384=12 ; JNPES_c384=6
+  TASKS_c384=480 ; TPN_c384=12 ; INPES_c384=12 ; JNPES_c384=6
```
Was this change to TPN_c384 here and on line 245 requested? This does not appear to be used in either of the modified tests.

aliabdolali replied:
This was asked by @junwang-noaa
